### ISSUE TITLE
Animated 3D models

### ISSUE THREAD

**Issue by Eke** [mapbox/mapbox-maps-ios#1109]
Hello.

Is there a way to have animated `puck3D`? Or the only way is custom implementation?

Thank you for great job.

**Comment by ZiZasaurus** [comment]
@eke, thank you for reaching out. At the moment this is not a supported feature of puck3D , however, you could try to accomplish this using a custom implementation and CADisplayLink.

**Comment by Eke** [comment]
@ZiZasaurus Hello, thank you for response. Are there any examples of that approach in the wild?

**Comment by ZiZasaurus** [comment]
@Eke we don't currently have any examples to point you to but we will consider this feature for a future release.

**Comment by astojilj** [comment]
@Eke ,
Could you please share some more details about what kind of animation you're interested in - rotating the car wheels along route or something else? Thanks.

**Comment by Eke** [comment]
@astojilj Yeah, that would be enough. I'm trying to make person 3D model walk on the map, something similar to Pokemon Go character moving his legs while moving.

**Comment by GG-beyond** [comment]
> @astojilj Yeah, that would be enough. I'm trying to make person 3D model walk on the map, something similar to Pokemon Go character moving his legs while moving.

Yes, I've met the question as you too，and how did you do it? look forward to your reply。

**Comment by OdNairy** [comment]
Please, take a look at [SceneKit Example](https://github.com/mapbox/mapbox-maps-ios/blob/main/Apps/Examples/Examples/All%20Examples/SceneKitExample.swift). It should be sufficient to add animated models on top of MapView.

**Comment by chocoford** [comment]
Hello? Any updates?
It is possible to have an animated puck3D now?
